### PR TITLE
Feature/manage client

### LINKED ISSUE #4: Add Ability to Associate Clients with Events on Dashboard
This will be done through the clients mac address.

### PR DESCRIPTION
**Pull request by SpBills**
Closes #4

### CODE REVIEW

**Comment by SpBills** [comment]
@kcouv I would like you to review this PR.

File: src/App.tsx [hunk 1]
```diff
@@ -1,21 +1,26 @@
...
+  const [isNewClient, setIsNewClient] = useState<boolean>(false);
+  const [isNewEvent, setIsNewEvent] = useState<boolean>(false);
   const [newEvent, setNewEvent] = useState<EventInput>();
-  const [allEvents, setAllEvents] = useState<Array<Event>>();
-
-  const [showEventForm, setShowEventForm] = useState(true);
+  const [isClientModalOpen, setIsClientModalOpen] = useState<boolean>(false);
+  const [isEventModalOpen, setIsEventModalOpen] = useState<boolean>(false);
+  const [allEvents, setAllEvents] = useState<Array<Event>>([]);
+  const [allClients, setAllClients] = useState<Array<Client>>([]);
```
Comment (kcouv) [inline src/App.tsx:23]: Great updates on names with respect to conventions used in terminal app

File: src/components/ClientDisplayTable.tsx [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+              <TableCell>
+                {current_state(client)}
+              </TableCell>
+            </TableRow>
+          )
+        })}
+      </TableBody>
+    </Table>
+  )
+}
```
Comment (kcouv) [inline src/components/ClientDisplayTable.tsx:54]: Love the use of mui tables here, also personally really diggin mui in general. Makes css so much easier

File: src/App.tsx [hunk 3]
```diff
@@ -24,70 +29,54 @@ function App() {
...
-          />
-          <Button variant="contained" className="sendID" type="submit">
-            Send ID
-          </Button>
-        </form>
-      </Card>
+      <Typography variant="h3">CU Event Scanner Admin Dashboard</Typography>
+
+      <ClientModal handleClose={() => setIsClientModalOpen(false)} handleSave={() => saveClient()} setClient={(c: ClientInput) => { setActiveClient(c) }} client={activeClient!} events={allEvents} open={isClientModalOpen} />
+      <EventModal handleClose={() => setIsEventModalOpen(false)} handleSave={() => saveEvent()} setEvent={(c: Event) => { setActiveEvent(c) }} event={activeEvent!} open={isEventModalOpen} />
```
Comment (kcouv) [inline src/App.tsx:73]: Cool use of modals here, makes main dashboard less cluttered with forms, and removes need for other pages

**Review by kcouv — APPROVED** [review]
Looks good to me, I like how the page functions so far

**Pull request merged**